File: codigo/soluciones/kat/cropeasy.py
T = int(input())
for tcase in range(1, T + 1):
    n, A, B, C, D, x, y, M = list(map(int, input().split()))
    trees = []
    for _ in range(n):
        trees.append((x, y))
        x = (A * x + B)%M
        y = (C * y + D)%M
    ans = 0
    for i in range(n):
        for j in range(i + 1, n):
            for k in range(j + 1, n):
                if (trees[i][0] + trees[j][0] + trees[k][0])%3 != 0 or (trees[i][1] + trees[j][1] + trees[k][1])%3 != 0:
                    continue
                ans += 1
    print(f'Case #{tcase}: {ans}')

File: codigo/soluciones/kat/cropeasy.java
import java.io.*;
import java.util.*;
class Cropeasy {
    private void _main() {
        Scanner sc = new Scanner(new BufferedInputStream(System.in));
        int T = sc.nextInt();
        for (int tcase = 1; tcase <= T; ++tcase) {
            int n = sc.nextInt();
            long A = sc.nextInt(), B = sc.nextInt(), C = sc.nextInt(), D = sc.nextInt(), x = sc.nextInt(), y = sc.nextInt(), M = sc.nextInt();
            long[] treesx = new long[n], treesy = new long[n];
            for (int i = 0; i < n; ++i) {
                treesx[i] = x;
                treesy[i] = y;
                x = (A * x + B)%M;
                y = (C * y + D)%M;
            }
            int ans = 0;
            for (int i = 0; i < n; ++i) {
                for (int j = i + 1; j < n; ++j) {
                    for (int k = j + 1; k < n; ++k) {
                        if ((treesx[i] + treesx[j] + treesx[k])%3 != 0 || (treesy[i] + treesy[j] + treesy[k])%3 != 0) continue;
                        ++ans;
                    }
                }
            }
            System.out.println("Case #" + tcase + ": " + ans);
        }
        sc.close();
    }
    public static void main(String[] args) {
        (new Cropeasy())._main();
    }
}
